packages/SwingSet/src/controller/initializeSwingset.js

```diff
@@ -327,7 +327,7 @@ export async function initializeSwingset(
 
   // Override the worker type if specified by the caller, to avoid having to
   // edit the config just to run everything under a different manager.
-  const defaultManagerType = SWINGSET_WORKER_TYPE || config.defaultManagerType;
+  const defaultManagerType = config.defaultManagerType || SWINGSET_WORKER_TYPE;
```

Comment:
I'm not convinced this priority change is the right thing (though I'm persuaded that the use case motivating it is a real one that needs to be addressed somehow), because I think it means that some stuff that now goes through CI twice to cover both local and xsnap flavors now won't because the test's config file would now override the environment-derived controls that the CI script employs, and that consequently we'll lose some important test coverage.  I might be able to be convinced otherwise, but you'll have to sell me.

Reply:
Good question. Here's the list of non-SwingSet test files in our tree which mention `defaultManagerType: 'xs-worker'`:

```
% rgjs defaultManagerType|grep test|grep -v SwingSet/test |grep xs-worker
ERTP/test/swingsetTests/basicFunctionality/test-basicFunctionality.js:  config.defaultManagerType = 'xs-worker';
ERTP/test/swingsetTests/splitPayments/test-splitPayments.js:  config.defaultManagerType = 'xs-worker';
governance/test/swingsetTests/committeeBinary/test-committee.js:  config.defaultManagerType = 'xs-worker';
governance/test/swingsetTests/contractGovernor/test-governor.js:  config.defaultManagerType = 'xs-worker';
notifier/test/test-publish-kit.js:    defaultManagerType: 'xs-worker',
sharing-service/test/swingsetTests/sharingService/test-sharing.js:  config.defaultManagerType = 'xs-worker';
spawner/test/swingsetTests/contractHost/test-contractHost.js:  config.defaultManagerType = 'xs-worker';
vats/test/upgrading/test-upgrade-contracts.js:    defaultManagerType: 'local', // Overridden in CI with SWINGSET_WORKER_TYPE=xs-worker
vats/test/upgrading/test-upgrade-vats.js:    defaultManagerType: 'local', // Overridden in CI with SWINGSET_WORKER_TYPE=xs-worker
zoe/test/swingsetTests/brokenContracts/test-crashingContract.js:  config.defaultManagerType = 'xs-worker';
zoe/test/swingsetTests/makeKind/test-makeKind.js:  config.defaultManagerType = 'xs-worker';
zoe/test/swingsetTests/offerArgs/test-offerArgs.js:  config.defaultManagerType = 'xs-worker';
zoe/test/swingsetTests/privateArgs/test-privateArgs.js:  config.defaultManagerType = 'xs-worker';
zoe/test/swingsetTests/runMint/test-runMint.js:  config.defaultManagerType = 'xs-worker';
zoe/test/swingsetTests/upgradeCoveredCall/test-coveredCall-service-upgrade.js:    defaultManagerType: 'xs-worker',
zoe/test/swingsetTests/zoe/test-zoe-upgrade.js:    defaultManagerType: 'xs-worker',
zoe/test/swingsetTests/zoe/test-zoe.js:  config.defaultManagerType = 'xs-worker';
```

and here's the list that mention `local`:
```
% rgjs defaultManagerType|grep test|grep -v SwingSet/test |grep local
ERTP/test/swingsetTests/ertpService/test-ertp-service-upgrade.js:    defaultManagerType: 'local',
inter-protocol/test/swingsetTests/fluxAggregator/test-fluxAggregator-service-upgrade.js:    defaultManagerType: 'local',
smart-wallet/test/swingsetTests/upgradeWalletFactory/test-walletFactory-service-upgrade.js:    defaultManagerType: 'local',
vats/test/bootstrapTests/supports.js:  config.defaultManagerType = 'local';
vats/test/upgrading/test-upgrade-contracts.js:    defaultManagerType: 'local', // Overridden in CI with SWINGSET_WORKER_TYPE=xs-worker
vats/test/upgrading/test-upgrade-vats.js:    defaultManagerType: 'local', // Overridden in CI with SWINGSET_WORKER_TYPE=xs-worker
```


The two "Overridden" comments (`vats/test/upgrading/test-upgrade-{contracts,vats}.js`) suggest that these tests probably want to be exercised both ways, and yeah, my change will reduce their test coverage.

Let's see, tests probably fall into one of the following categories:

* don't care
* want to be exercised by both `local` and `xs-worker`
* only work under `xs-worker` (they need metering, or snapshots)
* only work under `local` (they use `setup()` or some funky hybrid "steal the `syscall` object" thing)
  * these are probably only a single vat, so `vats.NAME.creationOptions.managerType` is sufficient, they don't need to use an all-vat `defaultManagerType`

We run most (all?) unit tests twice, once with the `SWINGSET_WORKER_TYPE=xs-worker` override, and once without any override. The last-priority default is `local` (because `initializeSwingset.js` has a `case undefined: config.defaultManagerType: 'local'`), so any test which omits `defaultManagerType` will get run in both ways.

So maybe what I should do is:

* try to figure out why each of the tests above expressed a preference
* if there doesn't seem to be a good reason, remove their `defaultManagerType`, so they'll get run both ways
* if the test clearly needs xsnap, retain the xs-worker, and accept that this test will be run twice in the same `xsnap` way (wasting CI time)
* it seems unlikely that any non-SwingSet tests will need `local`, but if so, retain their `defaultManagerType: 'local'`, and accept the same wasted CI time (but running it twice as `local`)

Reply:
moved to https://github.com/Agoric/agoric-sdk/pull/7527